cmd/restic/cmd_check.go

```diff
@@ -233,27 +264,44 @@ func runCheck(opts CheckOptions, gopts GlobalOptions, args []string) error {
 		}
 	}
 
-	doReadData := func(bucket, totalBuckets uint) {
-		packs := restic.IDSet{}
+	doSelectPacksByNumbers := func(bucket, totalBuckets uint) *restic.IDSet {
+		packs := &restic.IDSet{}
 		for pack := range chkr.GetPacks() {
 			// If we ever check more than the first byte
 			// of pack, update totalBucketsMax.
 			if (uint(pack[0]) % totalBuckets) == (bucket - 1) {
 				packs.Insert(pack)
 			}
 		}
-		packCount := uint64(len(packs))
+		return packs
+	}
 
-		if packCount < chkr.CountPacks() {
-			Verbosef(fmt.Sprintf("read group #%d of %d data packs (out of total %d packs in %d groups)\n", bucket, packCount, chkr.CountPacks(), totalBuckets))
-		} else {
-			Verbosef("read all data\n")
+	doSelectPacksByPercentage := func(percentage float64) *restic.IDSet {
```

Comment:
If this were a top-level function, it could be tested independently and runCheck would be shorter.

Reply:
I'm working on this. Just lacking the time to finish it right now.

Reply:
Fixed in https://github.com/restic/restic/pull/3038/commits/249e776e37f9d63d3d18683722ad97008de81122

Reply:
LGTM, feel free to mark as resolved (I can't do that, apparently).